### PR TITLE
[MRG+2] changes for MEP12/sphinx-gallery compliance

### PR DESCRIPTION
**Pull request by yinleon**
Regarding #7206

### CODE REVIEW

File: examples/event_handling/poly_editor.py [hunk 1]
```diff
@@ -1,4 +1,8 @@
 """
+===========
+Poly Editor
+==========
```
Comment (tacaswell) [inline examples/event_handling/poly_editor.py]: missing '='

File: examples/event_handling/resample.py [hunk 2]
```diff
@@ -1,13 +1,20 @@
...
 import matplotlib.pyplot as plt
 
-
 # A class that will downsample the data and recompute when zoomed.
 class DataDisplayDownsampler(object):
     def __init__(self, xdata, ydata):
         self.origYData = ydata
         self.origXData = xdata
-        self.ratio = 5
+        self.ratio = 50
```
Comment (tacaswell) [inline examples/event_handling/resample.py]: This seems a bit too aggressive.

**Comment by tacaswell** [comment]
Left 2 small comments, but otherwise looks good!

**Comment by tacaswell** [comment]
In the future, it is better to make PRs against some branch _other_ than your master branch.

**Comment by yinleon** [comment]
@tacaswell will do! My first commit to an OSS project, so apologies.

File: examples/axisartist/demo_axisline_style.py [hunk 3]
```diff
@@ -9,10 +16,14 @@
...
 
     for direction in ["xzero", "yzero"]:
+        # adds arrows at the ends of each axis
         ax.axis[direction].set_axisline_style("-|>")
+        
+        # adds X and Y-axis from the origin
         ax.axis[direction].set_visible(True)
 
     for direction in ["left", "right", "bottom", "top"]:
+        # hides boarders
```
Comment (anntzer) [inline examples/axisartist/demo_axisline_style.py]: borders

File: examples/event_handling/pipong.py [hunk 4]
```diff
@@ -1,7 +1,10 @@
-# A matplotlib based game of Pong illustrating one way to write interactive
-# animation which are easily ported to multiple backends
-# pipong.py was written by Paul Ivanov <http://pirsquared.org>
+"""
+====
+PONG
```
Comment (anntzer) [inline examples/event_handling/pipong.py]: "Pong" (I don't think it should be all caps).

File: examples/pylab_examples/dolphin.py [hunk 5]
```diff
@@ -1,3 +1,11 @@
+"""
+=======
+Dophins
```
Comment (anntzer) [inline examples/pylab_examples/dolphin.py]: dolphins

File: examples/axisartist/demo_axisline_style.py [hunk 6]
```diff
@@ -1,3 +1,10 @@
+"""
+================
+Axis Line Styles
```
Comment (anntzer) [inline examples/axisartist/demo_axisline_style.py]: Looks like most already converted examples only capitalize the first word (which I prefer too), so let's stay consistent (applies to this and all files below).

File: examples/axisartist/demo_axisline_style.py (hunk 3 above)
Comment (QuLogic) [inline examples/axisartist/demo_axisline_style.py]: boarders -> borders

File: examples/mplot3d/mixed_subplots_demo.py [hunk 7]
```diff
@@ -1,5 +1,8 @@
 """
-Demonstrate the mixing of 2d and 3d subplots.
+============================================
+Demonstrate the mixing of 2d and 3d subplots
+============================================
```
Comment (anntzer) [inline examples/mplot3d/mixed_subplots_demo.py]: blank line

File: examples/event_handling/pipong.py [hunk 8]
```diff
@@ -1,7 +1,10 @@
...
+====
 
+A matplotlib based game of Pong illustrating one way to write interactive animation which are easily ported to multiple backends pipong.py was written by Paul Ivanov <http://pirsquared.org>
```
Comment (QuLogic) [inline examples/event_handling/pipong.py]: matplotlib -> Matplotlib
animation -> animations
Add period at end of sentence.

Also, please wrap this line at 79 characters.

File: examples/event_handling/resample.py [hunk 9]
```diff
@@ -1,7 +1,14 @@
+"""
+===============
+Resampling Data
+===============
+
+Downsampling lowers the sample rate or sample size of a signal. In this tutorial, the signal is downsampled when the plot is adjusted through dragging and zooming.
```
Comment (QuLogic) [inline examples/event_handling/resample.py]: Wrap this line.

File: examples/mplot3d/mixed_subplots_demo.py (hunk 7 above)
Comment (QuLogic) [inline examples/mplot3d/mixed_subplots_demo.py]: Missing blank line after, I think.

File: examples/pylab_examples/bar_stacked.py [hunk 10]
```diff
@@ -1,3 +1,12 @@
+"""
+=================
+Stacked Bar Graph
+=================
+
+This is an example of creating a stacked bar plot with error bars using `plt.bar`.
+Note the parameters `yerr` used for error bars, and `bottom` to stack the women's bars on top of the men's bars.
```
Comment (QuLogic) [inline examples/pylab_examples/bar_stacked.py]: Lines need to be wrapper.

File: examples/event_handling/pipong.py (hunk 8 above)
Comment (anntzer) [inline examples/event_handling/pipong.py]: Here and below, wrap your lines at 79 characters (most editors should provide some option to help that).

File: examples/pylab_examples/dolphin.py (hunk 5 above)
Comment (QuLogic) [inline examples/pylab_examples/dolphin.py]: Dophins -> Dolphins

File: examples/pylab_examples/dolphin.py [hunk 11]
```diff
@@ -1,3 +1,11 @@
...
+=======
+
+This example shows how to draw, and manipulate shapes given verticies and nodes using the patches, path and transforms classes.
```
Comment (QuLogic) [inline examples/pylab_examples/dolphin.py]: Wrap line.

File: examples/pylab_examples/fill_between_demo.py [hunk 12]
```diff
@@ -1,3 +1,11 @@
+"""
+==============================
+Filling the area between lines
+==============================
+
+This example shows how to use fill_between() to color between lines based on user-defined logic.
```
Comment (QuLogic) [inline examples/pylab_examples/fill_between_demo.py]: Wrap line.

Comment (anntzer) [inline examples/pylab_examples/fill_between_demo.py]:
```
`fill_between`
```
(with backquotes) -- ultimately this should create a link to the API docs.

File: examples/pylab_examples/geo_demo.py [hunk 13]
```diff
@@ -1,3 +1,12 @@
+"""
+======================
+Geographic Projections
+======================
+
+This shows 4 possible projections using subplot.
+Matplotlib also supports the <a href='http://matplotlib.org/basemap/'>Basemaps Toolkit</a> for geographic projections.
```
Comment (QuLogic) [inline examples/pylab_examples/geo_demo.py]: Wrap line. Probably should mention Cartopy as well.

File: examples/pylab_examples/mri_demo.py [hunk 14]
```diff
@@ -1,4 +1,11 @@
-"""Displays an MRI image."""
+"""
+===
+MRI
+===
+
+This example illustrates how to read an image (of an MRI) into a numpy array, and display it in greyscale using `ax.imshow`.
```
Comment (QuLogic) [inline examples/pylab_examples/mri_demo.py]: numpy -> NumPy

Wrap line.

File: examples/event_handling/pipong.py (hunk 8 above)
Comment (anntzer) [inline examples/event_handling/pipong.py]: Here and below, use the rst syntax for hyperlinks (http://www.sphinx-doc.org/en/stable/rest.html#hyperlinks).

File: examples/pylab_examples/dolphin.py (hunk 11 above)
Comment (anntzer) [inline examples/pylab_examples/dolphin.py]: "vertices".

Please spell-check your docstrings.

File: examples/pylab_examples/mri_demo.py (hunk 14 above)
Comment (anntzer) [inline examples/pylab_examples/mri_demo.py]: just `imshow`

**Comment by anntzer** [comment]
Apparently I reviewed this at the same time as @QuLogic :-)

**Comment by tacaswell** [comment]
Apparently other reviewer can spell better than I can..

**Comment by yinleon** [comment]
thanks @anntzer and @QuLogic, you two are equally thorough :)

File: examples/event_handling/pipong.py [hunk 15]
```diff
@@ -1,7 +1,12 @@
...
-# pipong.py was written by Paul Ivanov <http://pirsquared.org>
-
+"""
+====
+Pong
+====
+
+A Matplotlib based game of Pong illustrating one way to write interactive 
+animations which are easily ported to multiple backends pipong.py was written 
+by <a href='http://pirsquared.org'>Paul Ivanov</a>.
```
Comment (NelleV) [inline examples/event_handling/pipong.py]: I am somewhat not surprised by the original author of that example…
ping @ivanov :D

File: examples/mplot3d/mixed_subplots_demo.py (hunk 7 above)
Comment (yinleon) [inline examples/mplot3d/mixed_subplots_demo.py]: I think I resolved this?

Comment (yinleon) [inline examples/mplot3d/mixed_subplots_demo.py]: I think I resolved this?

**Comment by yinleon** [comment]
should I be alarmed about the Travis CLI failure? Now sure I understand what I'm seeing under Details.

File: examples/event_handling/pipong.py [hunk 16]
```diff
@@ -1,7 +1,12 @@
-# A matplotlib based game of Pong illustrating one way to write interactive
-# animation which are easily ported to multiple backends
-# pipong.py was written by Paul Ivanov <http://pirsquared.org>
-
+"""
+====
+Pong
+====
+
+A Matplotlib based game of Pong illustrating one way to write interactive
```
Comment (afvincent) [inline examples/event_handling/pipong.py]: Beware of the trailing whitespace (remove it for PEP8-compliance)

File: examples/event_handling/pipong.py [hunk 17]
```diff
@@ -1,7 +1,12 @@
...
-# animation which are easily ported to multiple backends
-# pipong.py was written by Paul Ivanov <http://pirsquared.org>
-
+"""
+====
+Pong
+====
+
+A Matplotlib based game of Pong illustrating one way to write interactive 
+animations which are easily ported to multiple backends pipong.py was written
```
Comment (afvincent) [inline examples/event_handling/pipong.py]: As above, just remove the trailing whitespace for PEP8-compliance

**Comment by afvincent** [comment]
The Travis failure is related to PEP8 (more precisely to two trailing whitespace). More details [here](https://www.python.org/dev/peps/pep-0008/#other-recommendations) ;).

**Comment by QuLogic** [comment]
If you have changes to multiple files, editing on GitHub is not a good choice. It creates way too many single commits of one or two minor changes. Please rebase and squash this on your local copy if you can.

**Comment by NelleV** [comment]
Can we activate the squash and merge option? This would allow us to squash and merge before merging instead of asking contributors to rebase themselves.

**Comment by QuLogic** [comment]
TBH, as a _contributor_, I don't really like squash&merge as it decouples your fork and upstream's.

**Comment by yinleon** [comment]
Hi, I squashed all 20 commits. Did I do it correctly? Is it reflected in this PR, or should I make a new one?

**Comment by NelleV** [comment]
Yep, it is done and it works! Thanks!

**Comment by QuLogic** [comment]
> Hi, I squashed all 20 commits. Did I do it correctly? Is it reflected in this PR, or should I make a new one?

No, it's not quite right. After rebasing, you need to force push the result. If you don't force, git will complain and suggest you pull, but that will restore all the previous commits and add another merge commit. So, please do the rebase, and then do `git push --force origin master` (btw, it's safer to use a feature branch instead of `master`) so that you can _replace_ all the old commits.

**Comment by NelleV** [comment]
@yinleon Are you still around Berkeley? I was mistaken and something didn't quite happen right with the rebase. I am happy to help you out with this tomorrow (I should be at BIDS all afternoon if you are around).

**Comment by yinleon** [comment]
Sure thing @QuLogic.
@NelleV yes-- I will be at BIDS tomorrow.
Sorry for the confusion all.

File: examples/axisartist/demo_axisline_style.py [hunk 18]
```diff
@@ -9,10 +16,14 @@
     fig.add_subplot(ax)
 
     for direction in ["xzero", "yzero"]:
+        # adds arrows at the ends of each axis
         ax.axis[direction].set_axisline_style("-|>")
+
```
Comment (QuLogic) [inline examples/axisartist/demo_axisline_style.py]: There is some whitespace here causing the pep8 failure.

**Comment by NelleV** [comment]
Hi @yinleon 
There was a small blank line issue, that I fixed.

This patch needs an additional review.

File: examples/event_handling/poly_editor.py [hunk 19]
```diff
@@ -1,7 +1,10 @@
 """
-This is an example to show how to build cross-GUI applications using
-matplotlib event handling to interact with objects on the canvas
+===========
+Poly Editor
+==========
```
Comment (QuLogic) [inline examples/event_handling/poly_editor.py]: One missing `=`.

**Comment by NelleV** [comment]
I think this is ready to be merged. @anntzer can you have a look at this?

File: examples/event_handling/pipong.py [hunk 20]
```diff
@@ -1,7 +1,12 @@
...
-# animation which are easily ported to multiple backends
-# pipong.py was written by Paul Ivanov <http://pirsquared.org>
-
+"""
+====
+Pong
+====
+
+A Matplotlib based game of Pong illustrating one way to write interactive
+animations which are easily ported to multiple backends pipong.py was written
```
Comment (anntzer) [inline examples/event_handling/pipong.py]: dot after "multiple backends"

File: examples/event_handling/pipong.py [hunk 21]
```diff
@@ -1,7 +1,12 @@
...
+by <a href='http://pirsquared.org'>Paul Ivanov</a>.
```
Comment (anntzer) [inline examples/event_handling/pipong.py]: hyperlinks should use rst syntax (http://docutils.sourceforge.net/docs/ref/rst/restructuredtext.html#embedded-uris-and-aliases)

File: examples/mplot3d/lorenz_attractor.py [hunk 22]
```diff
@@ -1,6 +1,11 @@
 '''
-Plot of the Lorenz Attractor based on Edward Lorenz's 1963 "Deterministic
-Nonperiodic Flow" publication.
+================
+Lorenz Attractor
+================
+
+This is an example of plotting Edward Lorenz's 1963 "Deterministic
+Nonperiodic Flow" in a 3-dimensional space using mplot3d.
+
```
Comment (anntzer) [inline examples/mplot3d/lorenz_attractor.py]: I would suggest using an explicit hyperlink target (http://docutils.sourceforge.net/docs/ref/rst/restructuredtext.html#hyperlink-targets).

File: examples/pylab_examples/bar_stacked.py [hunk 23]
```diff
@@ -1,3 +1,13 @@
+"""
+=================
+Stacked Bar Graph
+=================
+
+This is an example of creating a stacked bar plot with error bars using
+`plt.bar`. Note the parameters `yerr` used for error bars, and `bottom`
```
Comment (anntzer) [inline examples/pylab_examples/bar_stacked.py]: yerr and bottom should use `*foo*`, not backticks (they are not references to linkable elements).

File: examples/pylab_examples/dolphin.py [hunk 24]
```diff
@@ -1,3 +1,12 @@
+"""
+========
+Dolphins
+========
+
+This example shows how to draw, and manipulate shapes given vertices and nodes
+using the `patches`, `path` and `transforms` classes.
```
Comment (anntzer) [inline examples/pylab_examples/dolphin.py]: `Patch`, `Path`, and `Transform` (references to actual python names).

File: examples/pylab_examples/geo_demo.py [hunk 25]
```diff
@@ -1,3 +1,15 @@
+"""
+======================
+Geographic Projections
+======================
+
+This shows 4 possible projections using subplot.
+Matplotlib also supports
+<a href='http://matplotlib.org/basemap/'>Basemaps Toolkit</a> and
```
Comment (anntzer) [inline examples/pylab_examples/geo_demo.py]: same comment as above regarding hyperlink syntax (either way works).

**Comment by dstansby** [comment]
This is going to take a bit of a rebase now sphinx-gallery is merged. I've assigned this to myself and will try to do it in the next day or so.

**Comment by choldgraf** [comment]
hey @yinleon , a lot has changed with MPL docs in the last few weeks :)

This means it'll be a bit more complicated to get this merged. Do you have time to do a live skype where you and I can spot-check these issues and get it back to a mergeable state?

**Comment by dstansby** [comment]
So I actually had a go at rebasing this last night, and gave up after a while. Quite a lot of the changes have already been made (titles), and there's only a couple of things that are now new unfortunately.

I think the easiest thing to do here would be to manually compare the PR diff and the current files, and just copy across anything new.

**Comment by choldgraf** [comment]
@phobson I think that's a good idea - let's see if @yinleon is interested in finishing this up and then we can make a plan from there

**Comment by tacaswell** [comment]
Closing in favor of #8860

**Pull request closed without merging**